assembly/src/parsers/u32_ops.rs

```diff
@@ -821,12 +842,30 @@ fn assert_and_push_u32_param(
     let value = parse_int_param(op, 1, lower_bound, u32::MAX)?;
     push_value(span_ops, Felt::new(value as u64));
 
-    // Assert both nmbers are u32.
+    // Assert both numbers are u32.
     span_ops.push(Operation::U32assert2);
 
     Ok(())
 }
 
+/// Pushes the first param of the `op` onto the stack. The param is not checked for
+/// belonging to u32.
+///
+/// This operation takes 1 VM cycle.
+fn push_param(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
+    let param_value = op.parts()[1];
+
+    // attempt to parse the parameter value as an integer
+    let value = match param_value.parse::<u64>() {
+        Ok(i) => i,
+        Err(_) => return Err(AssemblyError::invalid_param(op, 1)),
+    };
```

Comment:
Can we not use `parse_int_param()` function instead of this?

Reply:
`parse_int_param()` parses parameter as `u32` and also performs a check whether the number is within the boundaries. In case of `wrapping` and `overflowing` operation modes we need to parse parameter as `u64` and we don't need to perform any checks, so I thought creating such function was worth it.

Reply:
Ah - I see! Let's do the following:

- Rename the current `parse_int_param()` into `parse_u32_param()`.
- Rename your new function into `parse_int_param()`.